### PR TITLE
ENH: Use latest GTest

### PR DESCRIPTION
**Pull request by dzenanz**
## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Comment by dzenanz** [comment]
@bradking ghostflow never reports, and kwrobot makes repeated comments with errors on PRs. It started happening a few days ago, I believe.

**Comment by bradking** [comment]
Do: check

**Comment by hjmjohnson** [comment]
Do: check

**Review by blowekamp — APPROVED** [review]
Thanks for keeping it updated.

File: Modules/ThirdParty/GoogleTest/UpdateFromUpstream.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ readonly name="GoogleTest"
 readonly ownership="GoogleTest Upstream <googletestframework@googlegroups.com>"
 readonly subtree="Modules/ThirdParty/GoogleTest/src/itkgoogletest"
 readonly repo="https://github.com/google/googletest.git"
-readonly tag="release-1.12.1"
+readonly tag="main"
```
[the remark this answers is not in the document]
Comment (dzenanz) [reply Modules/ThirdParty/GoogleTest/UpdateFromUpstream.sh:11]: GTest recommends latest commit, see https://github.com/google/googletest/tree/main#live-at-head.

**Comment by dzenanz** [comment]
One commit is the update in the separate GTest root tree, one is a merge commit, and one is a regular commit. We cannot squash them without seriously shooting ourselves in the foot on the versioning front.

File: Modules/ThirdParty/GoogleTest/UpdateFromUpstream.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (N-Dekker) [reply Modules/ThirdParty/GoogleTest/UpdateFromUpstream.sh:11]: Thanks for the link, @dzenanz I read that that's what they call their "Live at Head" philosophy. But I wonder if it's really a good strategy for us as users. Do they want us to test each and every revision of their work? As they do seem to tag GTest quite regularly now, I would rather use a tagged version, which includes release notes, and might be more stable.

**Comment by N-Dekker** [comment]
> One commit is the update in the separate GTest root tree, one is a merge commit, and one is a regular commit. We cannot squash them without seriously shooting ourselves in the foot on the versioning front.

OK, thanks for your explanation 👍

File: Modules/ThirdParty/GoogleTest/UpdateFromUpstream.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (dzenanz) [reply Modules/ThirdParty/GoogleTest/UpdateFromUpstream.sh]: OK, I now switched to the tagged version.

**Review by N-Dekker — APPROVED** [review]
👍

**Pull request merged**